events/app.py

```diff
@@ -0,0 +1,113 @@
+import os
+import logging
+import datetime
+import pymongo
+# import pprint
+
+from flask import Flask, request, Response
+
+app = Flask(__name__)
+
+# connect to MongoDB Atlas database
+MONGODB_ATLAS_USERNAME = os.environ.get(
+    "MONGODB_ATLAS_USERNAME")
+MONGODB_ATLAS_PASSWORD = os.environ.get(
+    "MONGODB_ATLAS_PASSWORD")
+MONGODB_ATLAS_CLUSTER_ADDRESS = os.environ.get(
+    "MONGODB_ATLAS_CLUSTER_ADDRESS")
+PYMONGO_URI = "mongodb+srv://{}:{}@{}".format(
+    MONGODB_ATLAS_USERNAME,
+    MONGODB_ATLAS_PASSWORD,
+    MONGODB_ATLAS_CLUSTER_ADDRESS)
+# print("Pymongo URI: {}".format(PYMONGO_URI))
+client = pymongo.MongoClient(PYMONGO_URI)
+testDB = client.test
+eventsDB = client.eventsDB
+events_coll = eventsDB.all_events
+
+
+@app.route('/v1/add', methods=['POST'])
+def add_event():
+    # TODO(cmei4444): Get event information from request data
+    info = request.form['info']
+    # Add information into database
+    current_time = datetime.datetime.now()
+    build_event_info(info, current_time)
+    event = Event(info)
+    if not event.is_valid():
+        return Response(
+            status=400,
+            response="Event info was entered incorrectly.",
+        )
+
+    event.add_to_db()
+    return Response(
+        status=201,
+    )
+
+
+def build_event_info(info, time):
+    """Adds created_at time to event info dict"""
+    # MongoDB timestamp precision level, floor to milliseconds
+    time -= datetime.timedelta(0, 0, time.microsecond % 1000)
+    info['created_at'] = time
+    return info
+
+
+@app.route('/v1/edit/<event_id>', methods=['PUT'])
+def edit_event(event_id):
+    """Edit the event with the given id"""
+    pass
+
+
+@app.route('/v1/', methods=['GET'])
+def get_all_events():
+    """Return a list of all events currently in the DB"""
+    pass
+
+
+@app.route('/v1/search', methods=['GET'])
+def search_event():
+    """Search for the event with the given name in the DB"""
+    pass
+
+
+@app.route('/v1/<event_id>', methods=['PUT'])
+def get_one_event(event_id):
+    """Retrieve one event by event_id"""
+    pass
+
+
+class Event(object):
+    attributes = [
```

Comment:
What happens if someone inserts an `info` which has fields other than the ones in `attributes`?

Reply:
Should this be required_attributes, and you accept other fields as well, or is this supposed to be all the attributes allowed?